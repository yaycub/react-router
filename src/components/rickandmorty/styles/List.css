.List{
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 15px;

  ul{
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
    align-items: center;

    li{
      margin: 10px;
      border: 5px double black;
      border-radius: 5px;
      padding: 10px;


      a{
        font-family: Arial, Helvetica, sans-serif;
        font-weight: bold;
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        color: black;
        text-decoration: none;
        margin: 2px;
      }

      img{
        width: 15vw;
        border-radius: 5px;
      }
    }
  }

  span{
    margin: 5px;
    font-family: Arial, Helvetica, sans-serif;
  }
}