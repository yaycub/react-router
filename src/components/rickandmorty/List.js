import React from 'react';
import PropTypes from 'prop-types';
import ListItem from './ListItem';
import styles from './styles/List.css';
import { Link } from 'react-router-dom';

const List = ({ characterArr, onIncrement, onDecrement, page }) => {
  const characterEls = characterArr.map(character => {
    return (
      <li key={character.id}>
        <Link to={`/characters/${character.id}`}>
          <ListItem {...character} /> 
        </Link>
      </li>
    );
  });

  return (
    <>
      <section className={styles.List}>
        <div>
          <button onClick={onDecrement}>&#8592;</button>
          <span>{page}/25</span>
          <button onClick={onIncrement}>&#8594;</button>
        </div>
        <ul>
          {characterEls}
        </ul>
      </section>
    </>
  );
};

List.propTypes = {
  characterArr: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  })).isRequired,
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,
  page: PropTypes.number.isRequired
};

export default List;
