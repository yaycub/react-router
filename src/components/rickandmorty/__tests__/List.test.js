import React from 'react';
import { shallow } from 'enzyme';
import List from '../List';

describe('List Component', () => {
  it('should render a List', () => {
    const wrapper = shallow(<List characterArr={[{
      name: 'Person 1',
      image: 'Person1.jpeg',
      id: 1
    },
    {
      name: 'Person 2',
      image: 'Person2.jpeg',
      id: 2
    }
    ]} onIncrement={(() => 'hi')}
    onDecrement={(() => 'bye')} page={1} />);

    expect(wrapper).toMatchSnapshot();
  });
});
