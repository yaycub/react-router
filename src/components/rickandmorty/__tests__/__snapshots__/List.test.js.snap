// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component should render a List 1`] = `
<Fragment>
  <div>
    <button
      onClick={[Function]}
    >
      ←
    </button>
    <span>
      1
      /25
    </span>
    <button
      onClick={[Function]}
    >
      →
    </button>
  </div>
  <ul>
    <li
      key="1"
    >
      <ListItem
        id={1}
        image="Person1.jpeg"
        name="Person 1"
      />
    </li>
    <li
      key="2"
    >
      <ListItem
        id={2}
        image="Person2.jpeg"
        name="Person 2"
      />
    </li>
  </ul>
</Fragment>
`;
